packages/foundry/src/vault/adapter/compound/compoundV3/CompoundV3Adapter.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+
+import { AdapterBase, IERC20, IERC20Metadata, SafeERC20, ERC20, Math, IStrategy, IAdapter } from "../../abstracts/AdapterBase.sol";
+import { WithRewards, IWithRewards } from "../../abstracts/WithRewards.sol";
+import { ICToken, ICometRewarder, IGovernor, IAdmin } from "./ICompoundV3.sol";
+
+/**
+ * @title   CompoundV3 Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for CompoundV3 Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/beefyfinance/beefy-contracts/blob/master/contracts/BIFI/vaults/BeefyVaultV6.sol.
+ * Allows wrapping CompoundV3 Vaults with or without an active Booster.
+ * Allows for additional strategies to use rewardsToken in case of an active Booster.
+ */
+contract CompoundV3Adapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  /// @notice The Compound cToken contract
+  ICToken public cToken;
+
+  /// @notice The Compound Comet rewarder contract.
+  ICometRewarder public cometRewarder;
+
+  /// @notice Check to see if Compound liquidity mining is active on this market
+  bool public isActiveCompRewards;
+
+  /*//////////////////////////////////////////////////////////////
+                            INITIALIZATION
+    //////////////////////////////////////////////////////////////*/
+
+  error InvalidAsset(address asset);
+
+  /**
+   * @notice Initialize a new CompoundV3 Adapter.
+   * @param adapterInitData Encoded data for the base adapter initialization.
+   * @param compoundV3InitData Encoded data for the beefy adapter initialization.
+   * @dev `_cToken` - The underlying asset supplied to and wrapped by Compound.
+   * @dev This function is called by the factory contract when deploying a new vault.
+   */
+  function initialize(bytes memory adapterInitData, address, bytes memory compoundV3InitData) external initializer {
+    __AdapterBase_init(adapterInitData);
+
+    _name = string.concat("Popcorn CompoundV3", IERC20Metadata(asset()).name(), " Adapter");
+    _symbol = string.concat("popB-", IERC20Metadata(asset()).symbol());
+
+    (address _cToken, address _cometRewarder) = abi.decode(compoundV3InitData, (address, address));
+
+    cToken = ICToken(_cToken);
+    cometRewarder = ICometRewarder(_cometRewarder);
```

Comment:
I think we should use the configurator as registry to control the cToken address. I couldnt find any onchain path to confirm the comeRewarder though. Could you check this out?